registry/storage/purgeuploads.go

```diff
@@ -70,14 +69,9 @@ func getOutstandingUploads(ctx context.Context, driver storageDriver.StorageDriv
 	err = driver.Walk(ctx, root, func(fileInfo storageDriver.FileInfo) error {
 		filePath := fileInfo.Path()
 		_, file := path.Split(filePath)
-		if file[0] == '_' {
+		if strings.HasPrefix(file, "_") && fileInfo.IsDir() && file != "_uploads" {
```

Comment:
Wouldn't it be the same? Unless the [0] was some kind of premature optimization.

```suggestion
		if fileInfo.IsDir() && file != "_uploads" {
```

Reply:
No it's not, we skip directories starting with `_`  (reserved directories) except `_upload`